sydent/http/servlets/registerservlet.py

```diff
@@ -56,22 +59,34 @@ async def render_POST(self, request: Request) -> JsonDict:
                 "error": "matrix_server_name must be a valid Matrix server name (IP address or hostname)",
             }
 
-        result = await self.client.get_json(
-            "matrix://%s/_matrix/federation/v1/openid/userinfo?access_token=%s"
-            % (
-                matrix_server,
-                urllib.parse.quote(args["access_token"]),
-            ),
-            1024 * 5,
-        )
+        try:
+            result = await self.client.get_json(
+                "matrix://%s/_matrix/federation/v1/openid/userinfo?access_token=%s"
+                % (
+                    matrix_server,
+                    urllib.parse.quote(args["access_token"]),
+                ),
+                1024 * 5,
+            )
+        except (DNSLookupError, ConnectError, ResponseFailed) as e:
+            logger.warning("Unable to contact %s: %s", matrix_server, e)
+            request.setResponseCode(HTTPStatus.INTERNAL_SERVER_ERROR)
+            return {
+                "errcode": "M_UNKNOWN",
```

Comment:
We should probably start a constants module (or file an issue to start one...)

Reply:
Agreed. Filed https://github.com/matrix-org/sydent/issues/457 .